drivers/i2c/i2c_mspm0g3xxx.c

```diff
@@ -822,15 +863,21 @@ static const struct i2c_driver_api i2c_mspm0g3xxx_driver_api = {
 	PINCTRL_DT_INST_DEFINE(index);                                                             \
                                                                                                    \
 	INTERRUPT_INIT_FUNCTION_DECLARATION(index);                                                \
-                                                                                                   \
+												   \
+	BUILD_ASSERT(!(DT_INST_PROP_BY_PHANDLE(index, clocks, clock_frequency) % DT_INST_PROP(index, clock_frequency)),	\
```

Comment:
I know you just got told to put this into a build assert. I am okay with that!

An alternative approach could be to "round" down the clock frequency, so if e.g. you request `81 kHz` you'll end up with `80 kHz` (and maybe a log saying that the exact frequency couldn't be obtained).

But for all of our use cases, the build error is fine, so you can just leave this as-is.

Reply:
I think I am more comfortable with the idea of the user having to specify the exact rate that the bus should run at. We don't have a lot of logging with zephyr to begin with. And logging is only available when attaching a debug board on a specific MCU. So this does give us more direct feedback when we do something wrong.

Reply:
That's also fine by me. With these drivers, we should just consider the upstreamability as well. So the limited logging availability for us is not applicable to everyone else

Reply:
This if case got dropped since we now have the dt bitrate calculated in the config.